src/coreclr/gc/gc.cpp

```diff
@@ -3930,6 +3920,43 @@ void region_allocator::delete_region (uint8_t* region_start)
 
     leave_spin_lock();
 }
+
+void region_allocator::move_highest_free_regions (int64_t n, bool small_region_p, region_free_list to_free_list[count_free_region_kinds])
+{
+    assert (n > 0);
+
+    uint32_t* current_index = region_map_left_end - 1;
+    uint32_t* lowest_index = region_map_left_start;
+
+    while (current_index >= lowest_index)
+    {
+        uint32_t current_val = *current_index;
+        uint32_t current_num_units = get_num_units (current_val);
+        bool free_p = is_unit_memory_free (current_val);
+        if (!free_p && ((current_num_units == 1) == small_region_p))
+        {
+            uint32_t* index = current_index - (current_num_units - 1);
+            heap_segment* region = get_region_info (region_address_of (index));
+            if (heap_segment_allocated (region) == nullptr)
+            {
+                if (n >= current_num_units)
+                {
+                    n -= current_num_units;
+
+                    region_free_list::unlink_region (region);
+
+                    free_region_kind kind = region_free_list::get_region_kind (region);
+                    to_free_list[kind].add_region_front(region);
+                }
```

Comment:
it's slightly better to have a static method in region_free_list that takes care of both getting the kind and adding the region to the free list of that kind...just so that we have less coupling between region_allocator and the rest.

Reply:
Ok, and then get_region_kind can be private.